pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/HasUserAgedUpRoute.java

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.ddp.route;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants.PathParam;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.JdbiUmbrellaStudy;
+import org.broadinstitute.ddp.db.dao.StudyGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserProfileDao;
+import org.broadinstitute.ddp.json.AgeUpResponse;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.governance.Governance;
+import org.broadinstitute.ddp.model.governance.GovernancePolicy;
+import org.broadinstitute.ddp.model.user.UserProfile;
+import org.broadinstitute.ddp.pex.TreeWalkInterpreter;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+import java.util.Optional;
+
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+
+/**
+ * Returns the boolean value whether the user has agedUp or not
+ */
+@Slf4j
+@AllArgsConstructor
+public class HasUserAgedUpRoute implements Route {
+
+    @Override
+    public AgeUpResponse handle(Request request, Response response) {
+        String userGuid = request.params(PathParam.USER_GUID);
+        String studyGuid = request.params(PathParam.STUDY_GUID);
+
+        log.info("Attempting to retrieve whether the participant:{} agedUp in study:{}",
+                userGuid, studyGuid);
+
+        Boolean hasAgedUp = TransactionWrapper.withTxn(handle -> {
+            Optional<Governance> governance = handle.attach(UserGovernanceDao.class)
+                    .findGovernancesByParticipantAndStudyGuids(userGuid, studyGuid).findFirst();
+            if (governance.isEmpty()) {
+                return false;
+            }
+            String operatorGuid = governance.get().getProxyUserGuid();
+            Long studyId = handle.attach(JdbiUmbrellaStudy.class)
+                    .getIdByGuid(studyGuid)
+                    .orElseGet(() -> {
```

Comment:
[orElseGet](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseGet-java.util.function.Supplier-) should be used to return a default value if one is not present in the Optional- not solely to throw an error. That said, this is awkward since `haltError` throws on it's own.

I suggest changing this to `orElseThrow` (so things are closer in line with the intended use of the Optional type), and change 
```java
throw ResponseUtil.haltError(response, SC_NOT_FOUND, new ApiError(ErrorCodes.STUDY_NOT_FOUND, msg));
```

to
```java
return ResponseUtil.haltError(response, SC_NOT_FOUND, new ApiError(ErrorCodes.STUDY_NOT_FOUND, msg));
```

`haltError` will still be the one to throw but it's signature does return an exception. Still a bit awkward, but more in line with the intended use.